### PR TITLE
Lungs and hearts for mutant species

### PR DESCRIPTION
**Pull request by coiax**
- CPR cannot be performed by NOBREATHers, (or people without lungs)
- CPR is ineffective on NOBREATHers (or people without lungs). It can still be performed, but it will have no benefit because _they don't breathe_.
- NOBREATHers do not process any breath code
- NOBREATHers do not take oxygen/suffocation damage, and cannot lose breath.
- This means they do not take suffocation damage in crit. Instead, NOBREATH species in crit will take constant brute damage as their bodies fall apart. There is a new flag NOCRITDAMAGE which disables both this and regular oxyloss crit.
- NOBLOOD species have no hearts, and ignore heart attacks.
- If a NOBREATH species that has a heart undergoes a heart attack, the lack of circulation of fluids to their tissues causes tissue damage (brute) and a build up of body toxins (2 toxin chemical a tick).
- NOBREATH species can talk without lungs.
- Lexorin does not affect NOBREATH species.
- NOHUNGER species do not have appendices.
- full_heal now only regenerates organs with admin_revive = TRUE
- full_heal(admin_revive=TRUE) now replaces a heart if it's missing, and attempts to insert a mutant tongue first if a tongue is missing.
- Changing species removes any lungs, hearts or appendices if they are not needed in the new form.
### Technical stuff
- Adds a new macro CHECK_DNA_AND_SPECIES(C) which is passed a /mob/living/carbon/C, which will make the function return if the carbon lacks a DNA or species. Given that THEY SHOULD ALWAYS have them, this is about as arbitrary as a bunch of if statements with if(C.dna && C.dna.species && ... )
- [x] Alternative crit for NOBREATH species (golems, ash walkers, skeletons)
- [x] Removal of heart requirements from NOBLOOD species
### Bugs
- [x] Golems still have hearts and lungs
- [x] Fixed bug where a golem->human conversion would fail to give humans hearts and lungs
- [x] Skeletons and other NOHUNGER species still have appendixes, because #17700 isn't merged yet
- [x] A skeleton missing its tongue admin revived does not get a new skeleton tongue, but an ordinary one

:cl: coiax
rscdel: NOBREATH species (golems, skeletons, abductors, ash walkers, zombies) no longer have or require lungs. They no longer take suffocation/oxyloss damage, cannot give CPR and cannot benefit from CPR. Instead of suffocation damage, during crit they take gradual brute damage.
rscdel: NOBLOOD (golems, skeletons, abductors, slimepeople, plasmamen) species no longer have or require hearts.
tweak: NOBREATH species that require hearts still take damage from heart attacks, as their tissues die and the lack of circulation causes toxins to build up.
rscdel: NOHUNGER (plasmamen, skeletons) species no longer have appendixes.
/:cl:

### CODE REVIEW

**Comment by Man-Shroom-** [comment]
What about abductor ayys, those guys have no mouth and probably have so crazy bio-organ lung bullshit

**Comment by Razharas** [comment]
You made this branch out of hunger branch?

**Comment by Anonmare** [comment]
Ayys should breathe Nitrogen tbh

**Comment by Fox-McCloud** [comment]
One issue you will run into is that if the species NO_BREATHE is implemented as you've stated, here, then any oxygen damage they accumulate will be "locked" to that amount, and they wont' be able to undo it without chems.

Easily possible to correct for, just wanted to point out that quirk.

**Comment by coiax** [comment]
Ayys do have mouths and do get hungry though. Their tongues just aren't capable of generating speech normally.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -900,7 +904,17 @@
...
+				var/we_lung = M.getorganslot("lungs")
+
+				if(we_breathe && we_lung)
+					M.do_cpr(H)
+				else if(we_breathe && !we_lung)
+					M << "<span class='warning'>You have no lungs to breathe \
+						with, so cannot peform CPR.</span>"
+				else
+					M << "<span class='notice'>You do not breathe, so \
+						cannot perform CPR.</span>"
```
Comment (Core0verload) [inline code/modules/mob/living/carbon/human/species.dm]: Looks like borken english to me.

Comment (PKPenguin321) [inline code/modules/mob/living/carbon/human/species.dm]: Yes. Should say something like "Species that don't breathe can't give CPR!"

**Comment by coiax** [comment]
@Razharas no, but since it touches some of the same code, I think there is some duplication

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 2]
```diff
@@ -310,6 +318,9 @@
 
 /obj/item/bodypart/r_leg
 	name = "right leg"
+	desc = "You put your right leg in, your right leg out. In, out, in, out, \
+		shake it all about. And apparently then it detaches.\n\
+		The hokey cokey has certainly changed a lot since space colonisation."
```
Comment (Supermichael777) [inline code/modules/surgery/bodyparts/bodyparts.dm]: pokey

Comment (coiax) [inline code/modules/surgery/bodyparts/bodyparts.dm]: has different names depending on region

**Comment by PKPenguin321** [comment]
Undocumented changes? I see some shit about a wabbajack statue and really stupid description updates.

File: code/modules/surgery/bodyparts/bodyparts.dm (hunk 2 above)
Comment (Cheridan) [inline code/modules/surgery/bodyparts/bodyparts.dm]: >The hokey cokey (United Kingdom), hokey tokey (New Zealand) , or hokey pokey (United States, Canada, Ireland, Australia, the Caribbean, Mexico) is a participation dance with a distinctive accompanying tune and lyric structure.

You're being outvoted 7 to 1.

Comment (coiax) [inline code/modules/surgery/bodyparts/bodyparts.dm]: curses foiled again

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/mob/living/carbon/human/species.dm]: >it got merged without fixing this

**Pull request merged**